subprojects/core-api/src/main/java/org/gradle/api/artifacts/VersionConstraint.java

```diff
@@ -32,9 +31,8 @@
      * The preferred version of a module. The preferred version of a module can typically be upgraded during dependency resolution,
      * unless further constraints are added.
      *
-     * @return the baseline version, often referred to as the preferred version.
+     * @return the preferred version, or empty string if no preferred version specified. Never null.
      */
-    @Nullable
```

Comment:
This is very good! This was bothering me already; that there _might_ be a difference between `null` and `""`. A big :+1: for this and the next commit which streamline this.